components/adblock_rust_ffi/src/lib.rs

```diff
@@ -343,6 +343,22 @@ pub unsafe extern "C" fn filter_list_metadata_title(
     }
 }
 
+/// Returns the amount of time this filter list should be considered valid for,
+/// in hours. Defaults to 168 (i.e. 7 days) if unspecified by the
+/// `FilterListMetadata`.
+#[no_mangle]
+pub unsafe extern "C" fn filter_list_metadata_expires(metadata: *const FilterListMetadata) -> u16 {
+    use adblock::lists::ExpiresInterval;
+
+    const DEFAULT_EXPIRES_HOURS: u16 = 7 * 24;
+
+    match (*metadata).expires.as_ref() {
+        Some(ExpiresInterval::Days(d)) => 24 * *d as u16,
```

Comment:
I wonder do we really need to have the dedicated enum values for days and hours.

Reply:
That API is intended as an AST-like representation - i.e. "just" parsing without applying any application logic. Could be used to _construct_ a list header as well.